src/coreclr/vm/interpexec.cpp

```diff
@@ -29,6 +31,46 @@ LONG IgnoreCppExceptionFilter(PEXCEPTION_POINTERS pExceptionInfo, PVOID pv)
         : EXCEPTION_EXECUTE_HANDLER;
 }
 
+template<typename Obj, typename Method, typename Tuple, size_t... Is>
+auto CallWithArgumentPack(Obj* obj, Method method, Tuple& args, std::index_sequence<Is...>)
+-> decltype((obj->*method)(std::get<Is>(args)...))
+{
+    return (obj->*method)(std::get<Is>(args)...);
+}
+
+template<typename Obj, typename Method, typename... Args>
+auto CallWithSEHWrapper(Obj* obj, Method method, Args... args) -> decltype((obj->*method)(args...))
```

Comment:
Could we instead use a no-parameter lambda here (and capture the args into the state object)? Would avoid pulling in tuple and that machinery.

Reply:
One of the reasons I prefer not using the tuple is it violates our C++ guidelines for the repo.

I also think this implementation looks much cleaner and easier to understand at a glance.

Reply:
I don't see anything about tuple violating our C++ guidelines. I assume you may be referring to https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/clr-code-guide.md#-2114-limit-usage-of-standard-template-types-in-shipping-executables. My understanding was that this section was more about stuff that would have parts of the code in the lib. The tuple, at least in this usage case, doesn't require any such code.

We had bad experience with performance of lambdas in the past (@davidwrighton tried to change some huge macros in GC to lambdas), so I was worried that the compiler would not be able to eliminate the calls in this case. Fortunately, it is not that case here.

Reply:
The concern @jkotas has (and the reason for the rules) is about any code that may exist in the headers or any statically-linked portions. Anything in the standard library that is loaded at runtime from the target machine's standard library implementation is fine.

Reply:
tuple has code that runs that's defined in the headers, so it meets the definition AFAIK. In practice we'd never have an issue, but my understanding is that we'd rather be more strict here than trying to be more granular header-by-header.